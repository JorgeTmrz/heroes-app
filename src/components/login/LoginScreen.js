import React, { useContext } from 'react'
import { useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = () => {

    const lastPath = localStorage.getItem("lastPath") || "/"

    const history = useHistory();
    const {dispatch} = useContext(AuthContext);
    const handleClick = () => {
        dispatch({
            type: types.login,
            payload: {
                name: "Jorge"
            }
        })
        
        history.replace(lastPath);
    }

    return (
        <div className = "container mt-5">
            <h1>Login</h1>
            <hr/>
            <button 
                className = "btn btn-outline-primary"
                onClick = {handleClick}
            >
                Log In
            </button>
        </div>
    )
}
