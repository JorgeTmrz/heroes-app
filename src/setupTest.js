import "@testing-library/jest-dom";
import { configure } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
