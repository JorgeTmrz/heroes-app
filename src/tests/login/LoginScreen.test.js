import { mount } from "enzyme"
import { Router } from "react-router-dom"
import { AuthContext } from "../../auth/AuthContext"
import { LoginScreen } from "../../components/login/LoginScreen"
import "../../setupTest"
import { types } from "../../types/types"

describe('LoginScreen test cases', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: "Jorge",
          logged: false,
        },
      };
    
      const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }

    const wrapper = mount(
    <AuthContext.Provider value = {contextValue}>
        <Router history = {historyMock}>
            <LoginScreen />
        </Router>
    </AuthContext.Provider>
    )

    test('should display itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should do the dispatch and the navigation', () => {
        wrapper.find("button").simulate("click");

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: "Jorge"
            }})

        expect(historyMock.replace).toHaveBeenCalled();
    })
    
})
