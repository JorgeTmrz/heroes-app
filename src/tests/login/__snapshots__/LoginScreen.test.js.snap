// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen test cases should display itself correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Log In
  </button>
</div>
`;
